HiP-DataStore/Core/HistoryUtil.cs

```diff
@@ -27,19 +32,45 @@ public static class HistoryUtil
         /// The event stream is assumed to be consistent. If the stream is inconsistent (e.g. has a create event
         /// immediately followed by another create event), the behavior and resulting summary is undefined.
         /// </remarks>
-        public static async Task<HistorySummary> GetSummaryAsync(IEventStream eventStream, EntityId entityId)
+        public static async Task<HistorySummary> GetSummaryAsync(IEventStream eventStream, EntityId entityId, string userStoreBaseUrl, DataStoreAuthConfig dataStoreAuthConfig)
         {
             var enumerator = eventStream.GetEnumerator();
             var summary = new HistorySummary();
+            
+            var userService = new UsersClient(userStoreBaseUrl);
+
+            //we should get machine to machine access token and assign it to userService.Authorization
+            string accessToken = await Auth.GetAccessTokenAsync(dataStoreAuthConfig.Domain, dataStoreAuthConfig.Audience, dataStoreAuthConfig.ClientId, dataStoreAuthConfig.ClientSecret);
+
+            userService.Authorization = "Bearer "+ accessToken;
+            //get the details of all the users, so we contact the UserStore once instead of contacting it everytime for every change
+            var allUsers = await userService.GetAllAsync(new UserQueryArgs());                           
```

Comment:
If an exception happens during the request to Auth0 or the UserStore it should be handeled correctly. I think the request should still return the history information just without the username, if an exception happens.

Reply:
I added the exception handling logic, so in case there is a problem obtaining the Auth0 access token or a problem obtaining the users' details from UserStore, then the application can continue and show the changes without the changers' names.